packages/overlay/src/integrations/sentry/components/traces/TraceDetails.tsx

```diff
@@ -13,7 +13,11 @@ export default function TraceDetails() {
   const { traceId, spanId } = useParams();
   const [spanNodeWidth, setSpanNodeWidth] = useState<number>(50);
   const [query, setQuery] = useState<string>('');
-  const debouncedQuery = useDebounce(query, 200);
+  const [debouncedQuery, setDebouncedQuery] = useState<string>('');
```

Comment:
I liked the `useDebounce` hook usage. I guess we can bring it back in the form of `useDebouncedState` or something which simply returns a `debounce`d setter.

Reply:
Hi @BYK,

I just want to add a point here. With the current useDebounce hook, we are passing a callback function instead of setting state directly. I believe this approach is more versatile since it allows the hook to be used anywhere in the app for various actions, such as setting state, making API calls, or performing other actions.

What do you think?

Reply:
Just going to try using `lodash.debounce` and see how that goes :)